### PR TITLE
Move Media Indexing into a new module to simplify dependencies.

### LINKED ISSUE #14941: Can we downgrade DocumentFormat.OpenXML to 2.20.0
Can we downgrade `DocumentFormat.OpenXML` to 2.20.0 ?

I have developed an Excel import/export function that is based on ClosedXML
Unfortunately, it relies on DocumentFormat.OpenXML v2.16.0

While DocumentFormat.OpenXML v3.0.0 was optimized for.NET 8, I tested that all OC unit tests were still available after demoting it to version 2.x

### PR DESCRIPTION
**Pull request by MikeAlhayek**
Fix #14941

### CODE REVIEW

**Comment by MikeAlhayek** [comment]
@sebastienros look at the feature, do you like the features split?


- If you want to continue to index `.PDF` file you'll need to enable `OrchardCore.Media.Indexing.Pdf` feature.
- If you needed to enable indexing for text file with `.txt`, `.md` extensions, you'll needed `OrchardCore.Media.Indexing.Text` feature.
- If you needed to enable indexing for other extensions like (`.docx`, or `.pptx`), you'll needed `OrchardCore.Media.Indexing.MicrosoftOffice` feature.

File: src/OrchardCore.Modules/OrchardCore.Media.Indexing/Manifest.cs [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+)]
+
+[assembly: Feature(
+    Id = "OrchardCore.Media.Indexing",
+    Name = "Media Indexing",
+    Description = "Provides a way to index media files with common format in search providers.",
+    Dependencies =
+    [
+        "OrchardCore.Media"
+    ],
```
Comment (hishamco) [inline src/OrchardCore.Modules/OrchardCore.Media.Indexing/Manifest.cs:17]:
```suggestion
    Dependencies = [ "OrchardCore.Media" ],
```

File: src/OrchardCore.Modules/OrchardCore.Media.Indexing/Manifest.cs [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+)]
+
+[assembly: Feature(
+    Id = "OrchardCore.Media.Indexing.MicrosoftOffice",
+    Name = "Microsoft Office Media Indexing",
+    Description = "Provides a way to index Microsoft Word and Power Point files in search providers.",
+    Dependencies =
+    [
+        "OrchardCore.Media.Indexing"
+    ],
```
Comment (hishamco) [inline src/OrchardCore.Modules/OrchardCore.Media.Indexing/Manifest.cs:29]:
```suggestion
    Dependencies = [ "OrchardCore.Media.Indexing" ],
```

File: src/OrchardCore.Modules/OrchardCore.Media.Indexing/Manifest.cs [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+        "OrchardCore.Media"
+    ],
+    Category = "Search",
+    EnabledByDependencyOnly = true
+)]
+
+[assembly: Feature(
+    Id = "OrchardCore.Media.Indexing.MicrosoftOffice",
+    Name = "Microsoft Office Media Indexing",
+    Description = "Provides a way to index Microsoft Word and Power Point files in search providers.",
```
Comment (hishamco) [inline src/OrchardCore.Modules/OrchardCore.Media.Indexing/Manifest.cs:25]:
```suggestion
    Description = "Provides a way to index Microsoft Office files including Word and Power Point files in search providers.",
```

File: src/OrchardCore.Modules/OrchardCore.Media.Indexing/Manifest.cs [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+)]
+
+[assembly: Feature(
+    Id = "OrchardCore.Media.Indexing.Pdf",
+    Name = "PDF Media Indexing",
+    Description = "Provides a way to index PDF files in search providers.",
+    Dependencies =
+    [
+        "OrchardCore.Media.Indexing"
+    ],
```
Comment (hishamco) [inline src/OrchardCore.Modules/OrchardCore.Media.Indexing/Manifest.cs:40]:
```suggestion
    Dependencies = [ "OrchardCore.Media.Indexing" ],
```

File: src/OrchardCore.Modules/OrchardCore.Media.Indexing/Manifest.cs [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+)]
+
+[assembly: Feature(
+    Id = "OrchardCore.Media.Indexing.Text",
+    Name = "Text Media Indexing",
+    Description = "Provides a way to index common text files like (.txt and .md) in search providers.",
+    Dependencies =
+    [
+        "OrchardCore.Media.Indexing"
+    ],
```
Comment (hishamco) [inline src/OrchardCore.Modules/OrchardCore.Media.Indexing/Manifest.cs:51]:
```suggestion
    Dependencies =  [  "OrchardCore.Media.Indexing" ],
```

File: src/OrchardCore.Modules/OrchardCore.Media.Indexing/OrchardCore.Media.Indexing.csproj [hunk 6]
```diff
@@ -0,0 +1,31 @@
+<Project Sdk="Microsoft.NET.Sdk.Razor">
+
+  <PropertyGroup>
+    <AddRazorSupportForMvc>true</AddRazorSupportForMvc>
+    <!-- NuGet properties-->
+    <Title>OrchardCore Media Indexing</Title>
+    <Description>
+      $(OCCMSDescription)
+
+      The media module adds media indexing support.
```
Comment (hishamco) [inline src/OrchardCore.Modules/OrchardCore.Media.Indexing/OrchardCore.Media.Indexing.csproj:10]:
```suggestion
      The media module adds media indexing support for various files.
```

File: src/OrchardCore.Modules/OrchardCore.Media/Manifest.cs [hunk 7]
```diff
@@ -11,39 +11,31 @@
     Id = "OrchardCore.Media",
     Name = "Media",
     Description = "The media module adds media management support.",
-    Dependencies = new[] { "OrchardCore.ContentTypes" },
+    Dependencies =
+    [
+        "OrchardCore.ContentTypes"
+    ],
```
Comment (hishamco) [inline src/OrchardCore.Modules/OrchardCore.Media/Manifest.cs:17]:
```suggestion
    Dependencies = [ "OrchardCore.ContentTypes" ],
```

File: src/docs/releases/1.9.0.md [hunk 8]
```diff
@@ -2,6 +2,18 @@
...
+
+## Media Indexing
+
+Previously, `.PDF` file where auto indexed in the search providers (Elasticsearch, Lucene or Azure AI Search). Now, if you want to continue to index `.PDF` file you'll need to enable `OrchardCore.Media.Indexing.Pdf` feature.
+
+Additionally, if you needed to enable indexing for text file with `.txt`, `.md` extensions, you'll needed `OrchardCore.Media.Indexing.Text` feature.
+
+If you needed to enable indexing for other extensions like (`.docx`, or `.pptx`), you'll needed `OrchardCore.Media.Indexing.MicrosoftOffice` feature.
+
+## Change Logs
```
Comment (hishamco) [inline src/docs/releases/1.9.0.md:15]: Why?

**Review by hishamco — APPROVED** [review]
Please few notes and looks perfect!!

**Comment by hishamco** [comment]
This time I approved with a few comments to avoid your usual comment that I'm requesting change then I might have forgotten to approve :)

Good job bro

File: src/OrchardCore.Modules/OrchardCore.Media.Indexing/Manifest.cs (hunk 1 above)
Comment (MikeAlhayek) [reply src/OrchardCore.Modules/OrchardCore.Media.Indexing/Manifest.cs:17]: I don't like the one liner which is why I span it on multiple lines on purpose.

File: src/docs/releases/1.9.0.md (hunk 8 above)
Comment (MikeAlhayek) [reply src/docs/releases/1.9.0.md:15]: @sebastienros suggested that we split these into multiple features to allow users to specify which extensions they want to support.

Comment (hishamco) [reply src/docs/releases/1.9.0.md:15]: I know the intention of this PR, I mean why we have an empty Change Logs

File: src/OrchardCore.Modules/OrchardCore.Media.Indexing/Manifest.cs (hunk 1 above)
Comment (hishamco) [reply src/OrchardCore.Modules/OrchardCore.Media.Indexing/Manifest.cs:17]: It was there if I'm wrong, we usually span into multiple lines if we have more than one or two, anyhow I think we're ready to merge this IMHO

File: src/docs/releases/1.9.0.md (hunk 8 above)
Comment (MikeAlhayek) [reply src/docs/releases/1.9.0.md:15]: its not empty. under it there is a note about the new Azure Search AI

**Comment by MikeAlhayek** [comment]
@hishamco I split this into 2 projects as per @sebastienros original request.

File: src/OrchardCore.Modules/OrchardCore.Media.Indexing.Pdf/Manifest.cs [hunk 9]
```diff
@@ -0,0 +1,14 @@
+using OrchardCore.Modules.Manifest;
+
+[assembly: Module(
+     Name = "PDF Media Indexing",
```
Comment (hishamco) [inline src/OrchardCore.Modules/OrchardCore.Media.Indexing.Pdf/Manifest.cs:4]:
```suggestion
    Name = "PDF Media Indexing",
```

File: src/OrchardCore.Modules/OrchardCore.Media.Indexing/Startup.cs [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+public class Startup : StartupBase
+{
+    public override void ConfigureServices(IServiceCollection services)
+    {
+        services.AddScoped<IContentFieldIndexHandler, MediaFieldIndexHandler>();
+    }
+}
+
+[Feature("OrchardCore.Media.Indexing.Pdf")]
+public class PdfStartup : StartupBase
```
Comment (hishamco) [inline src/OrchardCore.Modules/OrchardCore.Media.Indexing/Startup.cs:16]: Why do we need to have `OrchardCore.Media.Indexing` while we creating  a separate modules for media indexing?

**Comment by Skrypt** [comment]
Just name this OpenXML intead of MicrosoftOffice. A Word or Excel document is not bound to be from Microsoft, it could be OpenOffice too.

**Review by hishamco — CHANGES_REQUESTED** [review]

File: src/OrchardCore.Modules/OrchardCore.Media.Indexing/Startup.cs (hunk 10 above)
Comment (Skrypt) [reply src/OrchardCore.Modules/OrchardCore.Media.Indexing/Startup.cs:16]: The whole folder can be removed.

Comment (MikeAlhayek) [reply src/OrchardCore.Modules/OrchardCore.Media.Indexing/Startup.cs:16]: yes. I removed it from the project but forgot to remove the folder. I'll clean that up and rename the module as @Skrypt suggested.

**Comment by MikeAlhayek** [comment]
@Skrypt @hishamco  clean it up.

**Comment by Skrypt** [comment]
You already did 😄

**Comment by hishamco** [comment]
I just hold my reply because this comment https://github.com/OrchardCMS/OrchardCore/issues/14941#issuecomment-1879697766

**Pull request merged**